cdap-master/src/main/java/co/cask/cdap/data/tools/Main.java

```diff
@@ -156,115 +234,85 @@ private Action parseAction(String action) {
     }
   }
 
-  private void performUpgrade(Injector injector) throws Exception {
-    // Upgrade system datasets
-    upgradeSystemDatasets(injector);
-
-    // Upgrade all user hbase tables
-    upgradeUserTables(injector);
+  private void performUpgrade() throws Exception {
+    LOG.info("Upgrading System and User Datasets ...");
+    DatasetUpgrader dsUpgrade = injector.getInstance(DatasetUpgrader.class);
+    dsUpgrade.upgrade();
+  }
 
-    // Upgrade all queue and stream tables.
-    QueueAdmin queueAdmin = injector.getInstance(QueueAdmin.class);
-    queueAdmin.upgrade();
+  public static void main(String[] args) throws Exception {
+    Main thisMain = new Main();
+    thisMain.startUp();
+    try {
+      thisMain.doMain(args);
+    } catch (Throwable t) {
+      LOG.info("Failed to upgrade ...");
```

Comment:
this should be an error log, and it should include the throwable that caused the failure.